lib/models/problems/problem.dart

```diff
@@ -3,39 +3,25 @@ library problem;
 import 'dart:convert';
 
 import 'package:built_collection/built_collection.dart';
+import 'package:crowdleague/utils/serializers.dart';
 import 'package:built_value/built_value.dart';
+import 'package:crowdleague/models/problems/problem_base.dart';
 import 'package:built_value/serializer.dart';
-import 'package:crowdleague/enums/problem_type.dart';
-import 'package:crowdleague/models/app/app_state.dart';
-import 'package:crowdleague/utils/serializers.dart';
-import 'package:meta/meta.dart';
 
 part 'problem.g.dart';
 
-abstract class Problem implements Built<Problem, ProblemBuilder> {
-  ProblemType get type;
-  String get message;
-  BuiltMap<String, Object> get info;
-  @nullable
-  String get trace;
-  @nullable
-  AppState get state;
+abstract class Problem implements ProblemBase, Built<Problem, ProblemBuilder> {
```

Comment:
Can you remind me - what is the purpose of providing a Problem class?  If it's to allow a developer to create a Problem without having to create a specific Problembase subtype, I'm wondering if something like "GeneralProblem" or "AllPurporseProblem"... maybe "NonSpecificProblem" or even "UnkownProblem" ... might be a good way to find a balance between the strictness of requiring a specific subtype every time but not encouraging the use of an uninformative general type. I'm more than happy if you wanna leave this for me to PR later if I ever get motivated to do it :-)

Reply:
I think that's a good suggestion. Fixed!
I've also added some documentation to the class for further clarity.